core/store/ledgerstore/tx_handler.go

```diff
@@ -578,7 +578,7 @@ func handleShardNotifyMsg(msg *xshard_types.XShardNotify, store store.LedgerStor
 	if gasConsume < neovm.MIN_TRANSACTION_GAS {
 		gasConsume = neovm.MIN_TRANSACTION_GAS
 	}
-	if tx.GasPrice > 0 {
+	if tx.GasPrice > 0 && msg.Contract != utils.ShardMgmtContractAddress && msg.Contract != utils.ShardStakeAddress {
```

Comment:
Can user contract remote-notify ShardMgmtContract/ShardStake?

Reply:
No，method ```checkInvoke``` at file ```smart_contract.go``` check this case, ```neovm contract cannot x-shard call native contract```